### PR TITLE
fix(create-cloudflare): bump vitest-pool-workers version on the templates

### LINKED ISSUE #7815: 🐛 BUG: npm test fails on fresh hello world create cloudflare project
### Which Cloudflare product(s) does this pertain to?

C3 (npm create cloudflare), Wrangler, Workers Vitest Integration

### What versions are you using?

Wrangler 3.103.2
C3 10.9.2
"@cloudflare/vitest-pool-workers": "^0.5.2",

### What operating system and version are you using?

Mac latest

### Please provide a link to a minimal reproduction

https://github.com/irvinebroque/vitest-breakpoint

### Describe the Bug

1. `npm create cloudflare@latest`
2. Hello World + TypeScript
3. `cd project directory`
4. `npm test`
5. Observe error

```
➜  vitest-breakpoint git:(main) npm run test

> vitest-breakpoint@0.0.0 test
> vitest


 DEV  v2.1.8 /Users/brendan/src/proj/vitest-breakpoint


⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Unhandled Errors ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯

Vitest caught 1 unhandled error during the test run.
This might cause false positive tests. Resolve unhandled errors to make sure your tests are not affected.

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Unhandled Error ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
Error: In project vitest.config.mts's configuration file wrangler.json, `compatibility_flags` must contain one of "nodejs_compat"/"nodejs_compat_v2".
Either one of these flags is required to use `@cloudflare/vitest-pool-workers`.
 ❯ buildProjectWorkerOptions node_modules/@cloudflare/vitest-pool-workers/dist/pool/index.mjs:1270:13
 ❯ buildProjectMiniflareOptions node_modules/@cloudflare/vitest-pool-workers/dist/pool/index.mjs:1382:47
 ❯ getProjectMiniflare node_modules/@cloudflare/vitest-pool-workers/dist/pool/index.mjs:1409:21
 ❯ Object.runTests node_modules/@cloudflare/vitest-pool-workers/dist/pool/index.mjs:1683:26
 ❯ executeTests node_modules/vitest/dist/chunks/resolveConfig.RxKrDli4.js:7680:5
 ❯ node_modules/vitest/dist/chunks/cli-api.C2yC_ESk.js:10798:9
 ❯ Vitest.runFiles node_modules/vitest/dist/chunks/cli-api.C2yC_ESk.js:10820:12
 ❯ Vitest.start node_modules/vitest/dist/chunks/cli-api.C2yC_ESk.js:10662:7
 ❯ startVitest node_modules/vitest/dist/chunks/cli-api.C2yC_ESk.js:11828:7

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯

 Test Files  no tests
      Tests  no tests
     Errors  1 error
   Start at  09:33:12
   Duration  407ms (transform 0ms, setup 0ms, collect 0ms, tests 0ms, environment 0ms, prepare 0ms)
```

refs https://github.com/cloudflare/workers-sdk/pull/7388 ?

**Pretty sure this is happening because `npm create cloudflare@latest` is not installing latest version of the vitest-pool-workers package**

### PR DESCRIPTION
**Pull request by edmundhung**
Fixes #7815.

---

- Tests
  - [ ] TODO (before merge)
  - [x] Tests included
  - [ ] Tests not necessary because:
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [x] Required
  - [ ] Not required because: 
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): 
  - [x] Documentation not necessary because: package version bump

### CODE REVIEW

File: packages/create-cloudflare/e2e-tests/workers.test.ts [hunk 1]
```diff
@@ -35,7 +36,20 @@ function getWorkerTests(opts: { experimental: boolean }): WorkerTestConfig[] {
 		return [
 			{
 				template: "hello-world",
-				variants: ["TypeScript", "JavaScript", "Python"],
```
[the remark this answers is not in the document]
Comment (edmundhung) [reply packages/create-cloudflare/e2e-tests/workers.test.ts:38]: I split the config to two as we have no test script for the python template. You can find the line for the python variant slightly below this line :)

[the remark this answers is not in the document]
Comment (CarmenPopoviciu) [reply packages/create-cloudflare/e2e-tests/workers.test.ts:38]: oh, that moved [here](https://github.com/cloudflare/workers-sdk/pull/7827/files#diff-e79a62fdea0c7ebd9090297979b8cd9fdfe82749e6497295ee4485c2d73daf7bR52)

File: packages/create-cloudflare/e2e-tests/helpers.ts [hunk 2]
```diff
@@ -79,6 +79,7 @@ export type RunnerConfig = {
 		route: string;
 		expectedText: string;
 	};
+	verifyTest?: boolean;
```
Comment (CarmenPopoviciu) [inline packages/create-cloudflare/e2e-tests/helpers.ts:82]: might be useful to describe what these `verify<X>`s do via a comment for each?

File: packages/create-cloudflare/e2e-tests/workers.test.ts [hunk 3]
```diff
@@ -35,7 +36,20 @@ function getWorkerTests(opts: { experimental: boolean }): WorkerTestConfig[] {
...
+				variants: ["TypeScript", "JavaScript"],
+				verifyDeploy: {
+					route: "/",
+					expectedText: "Hello World!",
+				},
+				verifyPreview: {
+					route: "/",
+					expectedText: "Hello World!",
+				},
+				verifyTest: true,
```
Comment (CarmenPopoviciu) [inline packages/create-cloudflare/e2e-tests/workers.test.ts:48]: this feels like something we should be doing for all C3 starter projects that come accompanied by tests?

Comment (edmundhung) [reply packages/create-cloudflare/e2e-tests/workers.test.ts:48]: Afaik only hello-world and the experimental hello-world-with-assets template (js/ts) includes some tests. But there is no e2e tests for non-framework experimental templates. Let me see if I can set it up for the assets template 👍🏼

File: packages/create-cloudflare/e2e-tests/helpers.ts (hunk 2 above)
Comment (edmundhung) [reply packages/create-cloudflare/e2e-tests/helpers.ts:91]: Added description on https://github.com/cloudflare/workers-sdk/pull/7827/commits/26d37a8b49762a209ac49e22496399d095b63f94

File: packages/create-cloudflare/e2e-tests/workers.test.ts (hunk 3 above)
Comment (edmundhung) [reply packages/create-cloudflare/e2e-tests/workers.test.ts:90]: Added tests for all non-framework experimental templates on https://github.com/cloudflare/workers-sdk/pull/7827/commits/68656739cf26984f8d21d4e235694982d42259d1 and https://github.com/cloudflare/workers-sdk/pull/7827/commits/c180e936b4e737d2c565bb302de7fc1e28c29705

Comment (CarmenPopoviciu) [reply packages/create-cloudflare/e2e-tests/workers.test.ts:94]: @edmundhung you are the abs best!!!! ♥️

File: packages/create-cloudflare/e2e-tests/helpers.ts [hunk 4]
```diff
@@ -71,14 +71,24 @@ export type RunnerConfig = {
...
 	};
+	/**
+	 * Specifies whether to run the preview script for the project and assert the response from the specified route.
+	 */
 	verifyPreview: null | {
 		route: string;
 		expectedText: string;
 	};
+	/**
+	 * Specifies whether to run the test script for the project and verify the exit code.
```
Comment (CarmenPopoviciu) [inline packages/create-cloudflare/e2e-tests/helpers.ts:89]: 🥹 sooo prity 🫶

File: packages/create-cloudflare/e2e-tests/helpers.ts [hunk 5]
```diff
@@ -25,7 +25,9 @@ import type {
 import type { Writable } from "stream";
 import type { RunnerTestCase, Suite, Test } from "vitest";
 
-export const C3_E2E_PREFIX = "tmp-e2e-c3";
+// Project Name prefix
+// Note: Project name can only be up to 58 characters long
+export const C3_E2E_PREFIX = "tmp-c3";
```
Comment (penalosa) [inline packages/create-cloudflare/e2e-tests/helpers.ts:30]: I think this will probably break the e2e cleanup scripts—is there a reason this needed to be changed?

Comment (edmundhung) [reply packages/create-cloudflare/e2e-tests/helpers.ts:30]: Good catch! That might explains why some e2e tests are failing 😅 

We are using `prefix + random suffix + template name + language` as the project name on the e2e tests and some of the templates end up with a project name more than 58 chars. So I was thinking if we can shrink the prefix a bit to save a few chars.. but maybe I should just trim it.

Comment (edmundhung) [reply packages/create-cloudflare/e2e-tests/helpers.ts]: Reverted in https://github.com/cloudflare/workers-sdk/pull/7827/commits/583d5510cd2f2583197962010e484fefee03dea2

File: packages/create-cloudflare/e2e-tests/workers.test.ts [hunk 6]
```diff
@@ -59,14 +119,14 @@ function getWorkerTests(opts: { experimental: boolean }): WorkerTestConfig[] {
...
-				variants: ["TypeScript", "JavaScript"],
+				variants: ["ts", "js"],
 				// Skipped for now, since C3 does not yet support resource creation
 				verifyDeploy: null,
 				verifyPreview: null,
 			},
 			{
 				template: "scheduled",
-				variants: ["TypeScript", "JavaScript"],
+				variants: ["ts", "js"],
```
Comment (edmundhung) [inline packages/create-cloudflare/e2e-tests/workers.test.ts:129]: All the variant name is changed as 
1. It is now passed as `--lang [lang]` instead of going through a prompt and have the test script figure out the right answer.
2. They are part of the project name and might become too long depending on the template name. It should be less likly to be over 58 chars now.

File: packages/create-cloudflare/e2e-tests/workers.test.ts [hunk 7]
```diff
@@ -109,15 +168,7 @@ describe
...
 								name: `${testConfig.name ?? testConfig.template}-${variant.toLowerCase()}`,
-								promptHandlers: [
-									{
-										matcher: /Which language do you want to use\?/,
-										input: {
-											type: "select",
-											target: variant,
-										},
-									},
-								],
```
Comment (edmundhung) [inline packages/create-cloudflare/e2e-tests/workers.test.ts:120]: We are now passing the `--lang` arg directly instead of trying to answer the prompt.

File: packages/create-cloudflare/e2e-tests/workers.test.ts [hunk 8]
```diff
@@ -137,9 +188,6 @@ describe
 							// Relevant project files should have been created
 							expect(project.path).toExist();
 
-							const gitignorePath = join(project.path, ".gitignore");
-							expect(gitignorePath).toExist();
-
```
Comment (edmundhung) [inline packages/create-cloudflare/e2e-tests/workers.test.ts:142]: The asset-only template has no `.gitignore`. I could have add an options to skip this only for that template but I think it is unnecessary to assume `.gitignore` exists on all template especially when the git setup on c3 is optional.

File: packages/create-cloudflare/e2e-tests/workers.test.ts [hunk 9]
```diff
@@ -151,22 +199,30 @@ describe
 
 							try {
 								expect(jsonPath).toExist();
-								const config = readJSON(jsonPath) as { main: string };
-								expect(join(project.path, config.main)).toExist();
-							} catch {
+								const config = readJSON(jsonPath) as { main?: string };
+								if (config.main) {
+									expect(join(project.path, config.main)).toExist();
+								}
```
Comment (edmundhung) [inline packages/create-cloudflare/e2e-tests/workers.test.ts:205]: The asset-only template has no main file 😅

File: packages/create-cloudflare/e2e-tests/helpers.ts [hunk 10]
```diff
@@ -372,9 +382,18 @@ export const testProjectDir = (suite: string, test: string) => {
 	const randomSuffix = crypto.randomBytes(4).toString("hex");
 	const baseProjectName = `${C3_E2E_PREFIX}${randomSuffix}`;
 
-	const getName = () =>
+	const getName = () => {
 		// Worker project names cannot be longer than 58 characters
-		`${baseProjectName}-${test.substring(0, 57 - baseProjectName.length)}`;
+		const projectName = `${baseProjectName}-${test.substring(0, 57 - baseProjectName.length)}`;
```
Comment (andyjessop) [inline packages/create-cloudflare/e2e-tests/helpers.ts:387]: [nit] Tiny edge case, because it's fine for now, but will this result in reusing the same name if `baseProjectName` is longer than 57 characters? We may change the way the base name is generated in the future.

**Review by andyjessop — APPROVED** [review]
